File: app/schemas/item.py
"""Item request/response schemas."""

from pydantic import BaseModel, Field
from datetime import date, datetime


class CreateItemRequest(BaseModel):
    title: str = Field(min_length=3, max_length=255)
    description: str | None = None
    category: str = Field(min_length=1, max_length=100)
    daily_rate: float = Field(gt=0)
    weekly_rate: float | None = Field(None, gt=0)
    deposit_required: float = Field(default=0.0, ge=0)
    condition_description: str | None = None
    image_urls: list[str] | None = None


class UpdateItemRequest(BaseModel):
    title: str | None = Field(None, min_length=3, max_length=255)
    description: str | None = None
    category: str | None = Field(None, min_length=1, max_length=100)
    daily_rate: float | None = Field(None, gt=0)
    weekly_rate: float | None = Field(None, gt=0)
    deposit_required: float | None = Field(None, ge=0)
    condition_description: str | None = None
    image_urls: list[str] | None = None
    status: str | None = None


class SetAvailabilityRequest(BaseModel):
    start_date: date
    end_date: date
    is_blocked: bool = False
    reason: str | None = None


class ItemResponse(BaseModel):
    id: str
    owner_id: str
    community_id: str
    title: str
    description: str | None = None
    category: str
    daily_rate: float
    weekly_rate: float | None = None
    deposit_required: float
    condition_description: str | None = None
    image_urls: list[str] | None = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class ItemSearchRequest(BaseModel):
    query: str | None = None
    category: str | None = None
    min_price: float | None = None
    max_price: float | None = None
    available_from: date | None = None
    available_to: date | None = None
    min_trust_score: float | None = None
    cursor: str | None = None
    limit: int = Field(default=20, ge=1, le=100)

File: neighborly_mobile/lib/features/items/models/item.dart
/// Item model matching the FastAPI ItemResponse schema.
class Item {
  final String id;
  final String ownerId;
  final String communityId;
  final String title;
  final String? description;
  final String category;
  final double dailyRate;
  final double? weeklyRate;
  final double depositRequired;
  final String? conditionDescription;
  final List<String>? imageUrls;
  final String status;
  final DateTime createdAt;

  const Item({
    required this.id,
    required this.ownerId,
    required this.communityId,
    required this.title,
    this.description,
    required this.category,
    required this.dailyRate,
    this.weeklyRate,
    required this.depositRequired,
    this.conditionDescription,
    this.imageUrls,
    required this.status,
    required this.createdAt,
  });

  String get primaryImage =>
      (imageUrls != null && imageUrls!.isNotEmpty) ? imageUrls!.first : '';

  bool get hasImages => imageUrls != null && imageUrls!.isNotEmpty;

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id'],
      ownerId: json['owner_id'],
      communityId: json['community_id'],
      title: json['title'],
      description: json['description'],
      category: json['category'],
      dailyRate: (json['daily_rate'] as num).toDouble(),
      weeklyRate: json['weekly_rate'] != null ? (json['weekly_rate'] as num).toDouble() : null,
      depositRequired: (json['deposit_required'] as num).toDouble(),
      conditionDescription: json['condition_description'],
      imageUrls: json['image_urls'] != null
          ? List<String>.from(json['image_urls'])
          : null,
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'owner_id': ownerId,
    'community_id': communityId,
    'title': title,
    'description': description,
    'category': category,
    'daily_rate': dailyRate,
    'weekly_rate': weeklyRate,
    'deposit_required': depositRequired,
    'condition_description': conditionDescription,
    'image_urls': imageUrls,
    'status': status,
    'created_at': createdAt.toIso8601String(),
  };
}
